### PR TITLE
Adds support for float and double types

### PR DESCRIPTION
**Pull request by atruskie**
This commit allows the CommandLineUtils to parse:

- double
- float
- and nullable variants of both

It also infers the CommandOptionType, in the same manner as is done for integers.

Support for parsing floating point numbers is important in, for example, CLI applications that process audio streams.

By the way, thanks for maintaining this library!

### CODE REVIEW

File: test/CommandLineUtils.Tests/ValueParserProviderTests.cs [hunk 1]
```diff
@@ -130,6 +144,76 @@ public void ParsesLongNullable(string arg, long? result)
...
+        [InlineData("0.0", 0.0f)]
+        [InlineData("123456789.987654321", 123456789.987654321f)]
+        [InlineData("-123.456", -123.456f)]
+        [InlineData("-1E10", -1E10f)]
+        [InlineData("-Infinity", float.NegativeInfinity)]
+        [InlineData("Infinity", float.PositiveInfinity)]
+        [InlineData("NaN", float.NaN)]
+        public void ParsesFloat(string arg, float result)
+        {
+            // can't work out how to set a string representation of infinity, in an attribute, for current culture
```
Comment (natemcmaster) [inline test/CommandLineUtils.Tests/ValueParserProviderTests.cs]: You probably would need to use `MemberData` instead, and return this from a property

Comment (atruskie) [reply test/CommandLineUtils.Tests/ValueParserProviderTests.cs]: Ah right. I've not used xunit before. I'll fix it up

Comment (natemcmaster) [reply test/CommandLineUtils.Tests/ValueParserProviderTests.cs]: Here's a good example:
```c#
public class CalculatorTests  
{
    [Theory]
    [MemberData(nameof(GetData), parameters: 3)]
    public void CanAddTheoryMemberDataMethod(int value1, int value2, int expected)
    {
        var calculator = new Calculator();

        var result = calculator.Add(value1, value2);

        Assert.Equal(expected, result);
    }

    public static IEnumerable<object[]> GetData(int numTests)
    {
        var allData = new List<object[]>
        {
            new object[] { 1, 2, 3 },
            new object[] { -4, -6, -10 },
            new object[] { -2, 2, 0 },
            new object[] { int.MinValue, -1, int.MaxValue },
        };

        return allData.Take(numTests);
    }
}
```
https://andrewlock.net/creating-parameterised-tests-in-xunit-with-inlinedata-classdata-and-memberdata/

**Comment by atruskie** [comment]
@natemcmaster I rewrote the commit to keep history clean

**Comment by natemcmaster** [comment]
Looks good. Thanks @atruskie!

**Pull request merged**